packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -191,22 +219,85 @@ export class Outbox {
 			this.attachFlowBatch.contentSizeInBytes >=
 			this.params.config.compressionOptions.minimumBatchSizeInBytes
 		) {
-			this.flushInternal(this.attachFlowBatch.popBatch());
+			this.flushInternal(this.attachFlowBatch);
 		}
 	}
 
 	public flush() {
-		this.flushInternal(this.attachFlowBatch.popBatch());
-		this.flushInternal(this.mainBatch.popBatch());
+		if (this.isContextReentrant()) {
```

Comment:
Should this be an assert? It might make sense (from code size POV) if we expect to never get here, and also if we get here, it (exception) is not lost and bubbles up somehow into logs.
If that removes the need to have one extra callback (closeContainer) - even better.

Reply:
An assert here doesn't bubble up in the logs. It depends on the client handling it, this bubbles up to the event handler so it will be somehow hidden.

It is technically possible to get there with immediate flush and calling ordersequentially. The thing is that I'd like for this to close the container as recursion in this context can create a small storm of ops if flushing happens and this should preferably be as loud as possible.